subprojects/docs/src/docs/userguide/console.adoc

```diff
@@ -0,0 +1,147 @@
+
+----
+BUILD SUCCESSFUL in 2m 10s
+411 actionable tasks: 381 executed, 30 up-to-date
+----
+
+[[sec:console_build_output]]
+==== Build output
+
+Build output is rendered right above the build progress bar. Typical output includes log messages, messages from the standard output and error streams, compile warnings, test output events and many more.
+
+Starting with version 4.0 of Gradle, the console output has been reduced dramatically. The console does not render every task with its build status anymore - it only renders output that is emitted during a build lifecycle phase or a task execution. Avoiding an abundance of information keeps output as readable as possible. Gradle grouped output originating from a specific context e.g. all warnings from a compilation task. Grouped output is especially useful in a project setup that enables parallel project execution as it prevents concurrent interleaved messages to be interleaved without clearly indicating its origin (see xref:parallel_execution[]).
+
+NOTE: To render console output with the formatting of pre-4.0 versions of Gradle, add the <<gradle_command_line,command line>> option `--console=plain`.
+
+The following console output shows grouped output for the configuration phase and the task `:compileJava`:
+
+----
+> Configure project :
+Configuring project version.
+Using a single directory for all classes from a source set. This behaviour has been deprecated and is scheduled to be removed in Gradle 5.0
+        at jmh_6ef4kuwkdiq4mwpxq9wis0xfk.run(/Users/foobar/dev/projects/gradle/gradle/jmh.gradle:28)
+
+> Task :compileJava
+Note: Some input files use unchecked or unsafe operations.
+Note: Recompile with -Xlint:unchecked for details.
+----
+
+The console does not wait until a unit of work is fully completed. It flushes the output to the console after a short amount of time to ensure that relevant information is made available as soon as possible. In a build environment that executes projects in a parallel the output could look as follows:
+
+
+----
+> Task :compileJava
+Note: Some input files use unchecked or unsafe operations.
+Note: Recompile with -Xlint:unchecked for details.
+
+> Task :generateCode
+Generating JAXB classed from XSD files.
+
+> Task :compileJava
+Note: Some input files use or override a deprecated API.
+Note: Recompile with -Xlint:deprecation for details.
+----
+
+[[sec:console_non_interactive_environments]]
+=== Look & feel in non-interactive environments
+
+By default Gradle tries to enable rich console output so that it can be color coded and apply additional formatting. Non-interactive environments fall back to plain console output. The plain output format does not support some of the rich visual elements mentioned above e.g. output grouping.
```

Comment:
By default, Gradle tries to enable rich console output by detecting the type of console the build is running from.  This enables color and additional console output formatting. Non-interactive environments fall back to using plain console output. The plain output format does not support some of the console output features mentioned above, like output grouping, yet.